.github/workflows/rust-lints.yml

```diff
@@ -3,39 +3,67 @@ name: Rust-Lint
 on:
   pull_request:
     paths:
-      - "packages/engine/**.rs"
-      - "packages/engine/rustfmt.toml"
-      - "packages/engine/Cargo.*"
+      - '**.rs'
+      - '**.toml'
+      - '**Cargo.*'
 
 defaults:
   run:
     shell: bash
 
+
 concurrency:
   group: ${{ github.head_ref }}
   cancel-in-progress: true
 
+env:
+  CARGO_TERM_COLOR: always
+
 jobs:
+  find-crates:
+    name: Find all crates
+    runs-on: ubuntu-latest
+    outputs:
+      crates: ${{ steps.find.outputs.crates }}
+    steps:
+      - name: Checkout source code
+        uses: actions/checkout@v2
+      - name: Find crates
+        id: find
+        run: echo "::set-output name=crates::$(find -name Cargo.toml -printf '%h\n' | sed 's:./::' | jq -R | jq -sc)"
+
   rustfmt:
     name: rustfmt
+    needs: find-crates
     runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        directory: ${{ fromJSON(needs.find-crates.outputs.crates) }}
+      fail-fast: false
+    env:
+      V8_PATH: ${{ github.workspace }}/.v8
     steps:
       - name: Checkout source code
         uses: actions/checkout@v2
+        
+        # Emitting checkstyle results in multiple xml documents for each crate so rustfmt is run for each crate
+      - name: Remove default members
+        run: |
+          sed '/default-members/d' -i ${{ matrix.directory }}/Cargo.toml
 
       - name: Install Rust
         uses: actions-rs/toolchain@v1
         with:
           profile: minimal
-          toolchain: nightly
+          toolchain: nightly-2021-11-24
```

Comment:
Is there any way we can easily pick this up from the toolchain config file?

Reply:
Also thought about this and it should be possible but I don't want to waste more time on this as of now. I'll add a TODO though. (Actually, the CI will fail, if this does not match)

Reply:
Makes sense, and agreed